src/tables.jl

```diff
@@ -126,25 +126,37 @@ uniquely identify rows in the `table` then an `ArgumentError` is throw. If `forc
 then the duplicate (non-unique) entries will be overwritten.
 """
 function populate!(A, table, value::Symbol; force=false)
-    # Use a BitArray mask to detect duplicates and error instead of overwriting.
-    mask = force ? falses() : falses(size(A))
-
-    for r in Tables.rows(table)
-        vals = Tuple(Tables.getcolumn(r, c) for c in dimnames(A))
-        inds = map(findindex, vals, axiskeys(A))
-
-        # Handle duplicate error checking if applicable
-        if !force
-            # Error if mask already set.
-            mask[inds...] && throw(ArgumentError("Key $vals is not unique"))
-            # Set mask, marking that we've set this index
-            setindex!(mask, true, inds...)
-        end
+    cols = Tables.columns(table)
 
-        # Insert our value into the data array
-        setindex!(A, Tables.getcolumn(r, value), inds...)
+    indices = CartesianIndex.(
+        map(dimnames(A)) do nm
+            col = Tables.getcolumn(cols, nm)
+            inds = axiskeys(A, nm)
+            result = zeros(Int, length(col))
+
+            for idx in pairs(inds)
+                result[last(idx) .== col] .= first(idx)
+            end
+
+            return result
+        end...
+    )
+
+    # Ugly logic for reporting duplicates
+    if !(force || allunique(indices))
+        seen = Dict{eltype(indices), nothing}()
```

Comment:
I'm not sure if there's a better way to do this, but it's more or less what `allunique` is doing. That being said, `allunique` seems to be faster which is why we still include it in the condition. Since we know we're gonna throw an error at this point I'm not sure how much we want to optimized this part of the function.